File: app/src/main/java/net/squanchy/search/SearchResults.java
package net.squanchy.search;

import com.google.auto.value.AutoValue;

import java.util.List;

import net.squanchy.schedule.domain.view.Event;
import net.squanchy.speaker.domain.view.Speaker;

@AutoValue
public abstract class SearchResults {

    public static SearchResults create(List<Event> events, List<Speaker> speakers) {
        return new AutoValue_SearchResults(events, speakers);
    }

    public abstract List<Event> events();

    public abstract List<Speaker> speakers();
}

File: app/src/main/java/net/squanchy/search/SearchResults.kt
package net.squanchy.search

import net.squanchy.schedule.domain.view.Event
import net.squanchy.speaker.domain.view.Speaker

data class SearchResults(
        val events: List<Event>,
        val speakers: List<Speaker>
) {
    val isEmpty: Boolean
        get() = events.isEmpty() && speakers.isEmpty()

    companion object {

        fun create(events: List<Event>, speakers: List<Speaker>) = SearchResults(events, speakers)
    }
}
